content/en/network_performance_monitoring/devices/guide/build-ndm-profile.md

```diff
@@ -0,0 +1,149 @@
+**Note**: For more details on device use cases, see [Networking hardware][4].
+
+### Metrics selection
+
+Next, decide the metrics to collect. Devices often expose thousands of metrics and OIDs that can span dozens of MIBs.
+
+Some guidelines to help you in this process:
+
+- Keep the number of metrics between 10 and 40.
+- Explore base profiles to see which ones could be applicable to the device.
+- Explore manufacturer-specific MIB files looking for metrics such as:
+    - General health: status gauges
+    - Network traffic: bytes in/out, errors in/out
+    - CPU and memory usage
+    - Temperature: temperature sensors, thermal condition
+    - Power supply
+    - Storage
+    - Field-replaceable units ([FRU][5])
+
+## Implementation
+
+Datadog recommends following this development workflow:
+
+- Add a profile
+- Write tests
+- Use simulation data
+
+### Add a profile
+
+Add a profile by creating a `.yaml` file with the `sysobjectid` and metrics, for example:
+
+```yaml
+sysobjectid: 1.3.6.1.4.1.232.9.4.10
+
+metrics:
+  - MIB: CPQHLTH-MIB
+    symbol:
+      OID: 1.3.6.1.4.1.232.6.2.8.1.0
+      name: cpqHeSysUtilLifeTime
+```
+
+**Note**: `sysobjectid` can be a wildcard pattern to match a sub-tree of devices, for example: `1.3.6.1.131.12.4.*`.
+
+### Write tests
+
+Add a unit test to `test_profiles.py` to verify the metric is successfully collected by the integration when the profile is enabled. Unit tests are used to prevent regressions and help with maintenance.
+
+{{< code-block lang="python" filename="test_profiles.py" wrap="false">}}
+def test_hp_ilo4(aggregator):
+    run_profile_check('hp_ilo4')
+    common_tags = common.CHECK_TAGS + ['snmp_profile:hp-ilo4']
+    aggregator.assert_metric('snmp.cpqHeSysUtilLifeTime', metric_type=aggregator.MONOTONIC_COUNT, tags=common_tags, count=1)
+    aggregator.assert_all_metrics_covered()
+{{< /code-block >}}
+
+If you run this test without simulation data yet, it fails:
+
+{{< code-block lang="console" disable_copy="true" wrap="false">}}
+$ ddev test -k test_hp_ilo4 snmp:py38
+[...]
+======================================= FAILURES ========================================
+_____________________________________ test_hp_ilo4 ______________________________________
+tests/test_profiles.py:1464: in test_hp_ilo4
+    aggregator.assert_metric('snmp.cpqHeSysUtilLifeTime', metric_type=aggregator.GAUGE, tags=common.CHECK_TAGS, count=1)
+../datadog_checks_base/datadog_checks/base/stubs/aggregator.py:253: in assert_metric
+    self._assert(condition, msg=msg, expected_stub=expected_metric, submitted_elements=self._metrics)
+../datadog_checks_base/datadog_checks/base/stubs/aggregator.py:295: in _assert
+    assert condition, new_msg
+E   AssertionError: Needed exactly 1 candidates for 'snmp.cpqHeSysUtilLifeTime', got 0
+[...]
+{{< /code-block >}}
+
+### Add simulation data
+
+Add a `.snmprec` file named after the `community_string`. This is the same value given to `run_profile_check()` in the example above:
```

Comment:
It's better to recommend generating the `.snmprec`, see `Generate table simulation data`:

https://datadoghq.dev/integrations-core/tutorials/snmp/how-to/#generate-table-simulation-data

We sometimes modify the `.snmprec`, but only to tweak / finetune.

Reply:
Removing this section in favor of testing with an actual device